src/chunk/soc.ts

```diff
@@ -0,0 +1,245 @@
+import { Bytes, FlexBytes, verifyBytes, verifyFlexBytes } from './bytes'
+import { bmtHash } from './bmt'
+import { EthAddress, recoverAddress, sign, Signature, Signer } from './signer'
+import { keccak256Hash } from './hash'
+import { makeSpan } from './span'
+import { serializeBytes } from './serialize'
+import { BeeError } from '../utils/error'
+import { BrandedType } from '../types'
+
+const MIN_PAYLOAD_SIZE = 1
+const MAX_PAYLOAD_SIZE = 4096
+
+const SPAN_SIZE = 8
+const IDENTIFIER_SIZE = 32
+const SIGNATURE_SIZE = 65
+
+const CAC_SPAN_OFFSET = 0
+const CAC_PAYLOAD_OFFSET = CAC_SPAN_OFFSET + SPAN_SIZE
+
+const SOC_IDENTIFIER_OFFSET = 0
+const SOC_SIGNATURE_OFFSET = SOC_IDENTIFIER_OFFSET + IDENTIFIER_SIZE
+const SOC_SPAN_OFFSET = SOC_SIGNATURE_OFFSET + SIGNATURE_SIZE
+const SOC_PAYLOAD_OFFSET = SOC_SPAN_OFFSET + SPAN_SIZE
+
+type ChunkAddress = Bytes<32>
+
+/**
+ * General chunk interface for Swarm
+ *
+ * It stores the serialized data and provides functions to access
+ * the fields of a chunk.
+ *
+ * It also provides an address function to calculate the address of
+ * the chunk that is required for the Chunk API.
+ */
+export interface Chunk {
+  readonly data: Uint8Array
+  span(): Bytes<8>
+  payload(): FlexBytes<1, 4096>
+
+  address(): ChunkAddress
+}
+
+export type Identifier = Bytes<32>
+
+/**
+ * With single owner chunks, a user can assign arbitrary data to an
+ * address and attest chunk integrity with their digital signature.
+ *
+ * This interface extends the Chunk interface so it has the same
+ * properties, but the address calculation is based on the identifier
+ * and the owner of the chunk.
+ */
+export interface SingleOwnerChunk extends Chunk {
+  identifier: () => Identifier
+  signature: () => Signature
+}
+
+type ValidChunkData = BrandedType<Uint8Array, 'ValidChunkData'>
+type ValidSingleOwnerChunkData = BrandedType<Uint8Array, 'ValidSingleOwnerChunkData'>
+
+type SpanReference = Bytes<32> | Bytes<64>
+
+/**
+ * Creates a content addressed chunk and verifies the payload size.
+ *
+ * @param payloadBytes the data to be stored in the chunk
+ */
+export function makeContentAddressedChunk(payloadBytes: Uint8Array): Chunk {
+  const span = makeSpan(payloadBytes.length)
+  const payload = verifyFlexBytes(MIN_PAYLOAD_SIZE, MAX_PAYLOAD_SIZE, payloadBytes)
+  const data = serializeBytes(span, payload) as ValidChunkData
```

Comment:
More consistent with other usage and less magic

```suggestion
  verifyFlexBytes(MIN_PAYLOAD_SIZE, MAX_PAYLOAD_SIZE, payloadBytes)
  const data = serializeBytes(span, payloadBytes) as ValidChunkData
```